posthog/api/test/test_feature_flag.py

```diff
@@ -1086,6 +1089,207 @@ def test_local_evaluation(self, mock_capture):
 
         self.assertEqual(response_data["group_type_mapping"], {"0": "organization", "1": "company"})
 
+    @patch("posthog.api.feature_flag.report_user_action")
+    def test_evaluation_reasons(self, mock_capture):
+        FeatureFlag.objects.all().delete()
+        GroupTypeMapping.objects.create(team=self.team, group_type="organization", group_type_index=0)
+        GroupTypeMapping.objects.create(team=self.team, group_type="company", group_type_index=1)
+        Person.objects.create(
+            team_id=self.team.pk,
+            distinct_ids=["1", "2"],
+            properties={"beta-property": "beta-value"},
+        )
+
+        self.client.post(
+            f"/api/projects/{self.team.id}/feature_flags/",
+            {
+                "name": "Alpha feature",
+                "key": "alpha-feature",
+                "filters": {
+                    "groups": [{"rollout_percentage": 20}],
+                    "multivariate": {
+                        "variants": [
+                            {"key": "first-variant", "name": "First Variant", "rollout_percentage": 50},
+                            {"key": "second-variant", "name": "Second Variant", "rollout_percentage": 25},
+                            {"key": "third-variant", "name": "Third Variant", "rollout_percentage": 25},
+                        ],
+                    },
+                },
+            },
+            format="json",
+        )
+
+        self.client.post(
+            f"/api/projects/{self.team.id}/feature_flags/",
+            {
+                "name": "Group feature",
+                "key": "group-feature",
+                "filters": {
+                    "aggregation_group_type_index": 0,
+                    "groups": [{"rollout_percentage": 61}],
+                },
+            },
+            format="json",
+        )
+
+        # old style feature flags
+        FeatureFlag.objects.create(
+            name="Beta feature",
+            key="beta-feature",
+            team=self.team,
+            rollout_percentage=81,
+            filters={"properties": [{"key": "beta-property", "value": "beta-value"}]},
+            created_by=self.user,
+        )
+        # and inactive flag
+        FeatureFlag.objects.create(
+            name="Inactive feature",
+            key="inactive-flag",
+            team=self.team,
+            active=False,
+            rollout_percentage=100,
+            filters={"properties": []},
+            created_by=self.user,
+        )
+
+        # general test
+        response = self.client.get(
+            f"/api/projects/{self.team.pk}/feature_flags/evaluation_reasons",
+            {
+                "distinct_id": "test",
+            },
+        )
+        self.assertEqual(response.status_code, status.HTTP_200_OK)
+        response_data = response.json()
+        self.assertEqual(len(response_data), 4)
+
+        self.assertEqual(
+            response_data,
+            {
```

Comment:
Is there a reason we're returning the values in an object key value pairing instead of an array of objects? 
(I can update this just let me know if you want me to)

Reply:
Easier to add new things to it in the future 😅 , and I think makes sense here since we need a way to demarcate what's the `key`, and corresponding values.